src/puppetlabs/puppetdb/queue.clj

```diff
@@ -22,7 +26,24 @@
       (throw+ {:kind ::parse-error} e "Error parsing command"))))
 
 (defn metadata-str [received command version certname]
-  (format "%s_%s_%s_%s" (tcoerce/to-long received) command version certname))
+  (format "%s_%s_%s_%s.json" (tcoerce/to-long received) command version certname))
```

Comment:
I think this is causing the .json to get doubled up when there are preexisting messages in the queue -- starting with a nonempty queue will spew a bunch of file not found errors.